src/WebJobs.Script/azurefunctions/functions.js

```diff
@@ -127,3 +128,27 @@ function getEntryPoint(f, context) {
 
     return f;
 }
+
+function stripFunctions(val) {
+    return stripFunctionsDepth(val, 2);
+}
+
+// strip functions from 'val' nested up to 'depth' within arrays and objects
+// ignores non-objects and buffers
+function stripFunctionsDepth(val, depth) {
```

Comment:
I'm thinking we should do this in our .NET pipeline. We're using `ExpandoObjectConverter` already for deserialiation/serialization of `ExpandoObject`s we get from Node. I'm surprised that the base class doesn't handle funcs already, but I think we can just create a subclass that skips them:

```csharp
private class ScriptExpandoObjectJsonConverter : ExpandoObjectConverter
{
    public ScriptExpandoObjectJsonConverter()
    {
    }

    public override bool CanConvert(Type objectType)
    {
        if (typeof(Delegate).IsAssignableFrom(objectType))
        {
            return false;
        }
        return base.CanConvert(objectType);
    }
}
```

That will recursively handle funcs in returned objects. We might need some special handling for cases where just a func is returned, etc.